src/cljs/main/broadfcui/page/method_repo/methods_configs_acl.cljs

```diff
@@ -22,77 +23,80 @@
-           [:div {:style {:paddingBottom "0.5em" :fontSize "90%"}}
-            [:div {:style {:float "left" :width column-width}} "User or Group ID"]
-            [:div {:style {:float "right" :width column-width}} "Access Level"]
-            (common/clear-both)]
-           (map-indexed
-            (fn [i acl-entry]
-              [:div {}
-               [input/TextField
-                {:ref (str "acl-key" i)
-                 :style {:float "left" :width column-width
-                         :backgroundColor (when (< i (:count-orig @state))
-                                            (:background-light style/colors))}
-                 :disabled (< i (:count-orig @state))
-                 :spellCheck false
-                 :defaultValue (:user acl-entry)
-                 :predicates [(input/valid-email-or-empty "User ID")]}]
-               (style/create-identity-select
-                {:ref (str "acl-value" i)
-                 :style {:float "right" :width column-width :height 33}
-                 :defaultValue (:role acl-entry)}
-                access-levels)
-               (common/clear-both)])
-            (:acl-vec @state))
-           [comps/Button {:text "Add new" :icon :add-new
-                          :onClick #(swap! state assoc :acl-vec
-                                           (conj (react/call :capture-ui-state this)
-                                                 {:user "" :role reader-level}))}]
-           [:label {:style {:cursor "pointer"}}
-            [:input {:type "checkbox" :ref "publicbox"
-                     :style {:marginLeft "2em" :verticalAlign "middle"}
-                     :onChange #(swap! state assoc :public-status (-> (@refs "publicbox") .-checked))
-                     :checked (:public-status @state)}]
-            [:span {:style {:paddingLeft 6 :verticalAlign "middle"}} "Publicly Readable?"]]
-           (style/create-validation-error-message (:validation-error @state))
-           [comps/ErrorViewer {:error (:save-error @state)}]]
-          (:error @state) (style/create-server-error-message (cond (= (:error @state) "Forbidden") (str "You are unauthorized to edit this " (clojure.string/lower-case (:entityType props)) ".")
-                                                                   :else (:error @state)))
-          :else [comps/Spinner {:text
-                                (str "Loading Permissions for " (:title props) "...")}]))
-       :ok-button (when (:acl-vec @state) {:text "Save" :onClick #(react/call :persist-acl this)})}])
+        (net/render-with-ajax
+         (:acl-response @state)
+         #(this :-render-acl-form)
+         {:loading-text (str "Loading Permissions for " (:title props) "...")
+          :rephrase-error
+          #(net/create-error-message-for-code
+            (str "You are unauthorized to edit this "
+                 (clojure.string/lower-case (:entityType props)) ".")
+            403 %)}))
```

Comment:
This would become
```cljs
#(if (= 403 (:status-code %))
    (str "You are not...")
    (get-in % [:parsed-response :message]))
```
One less level of indirection, plus the status code is read from the XHR object instead of the parsed response, which is code that we wrote so may be less reliable.